src/Raven.Server/Rachis/Commands/Leader.LeaderModifyTopologyCommand.cs

```diff
@@ -106,11 +106,7 @@ protected override long ExecuteCmd(ClusterOperationContext context)
                 _engine.GetStateMachine().EnsureNodeRemovalOnDeletion(context, _leader.Term, _nodeTag);
             }
 
-            // after commit but still under the lock
-            context.Transaction.InnerTransaction.LowLevelTransaction.AfterCommitWhenNewTransactionsPrevented += (tx) =>
-            {
-                AfterCommit(index);
-            };
+            CompleteTopologyModificationAfterRachisCommit(index);
```

Comment:
this is not after the commit now?

Reply:
This happens after the commit, but we aren't involving Voron.
After the command executes, there is code to complete those tasks, and we'll run then.
See: https://github.com/ravendb/ravendb/blob/735df03075c401aa3fc1fd04ca1f19a3de883696/src/Raven.Server/Rachis/Leader.cs#L522